File: sdks/php-agent/src/Transport/GrpcTransport.php
<?php
declare(strict_types=1);

namespace HMS\CarHire\Transport;

use HMS\CarHire\Config;
use HMS\CarHire\Exceptions\TransportException;

/**
 * gRPC transport – STUBS until you run `composer run proto:gen` and wire service clients.
 * Implement methods by calling generated stubs with per-call deadlines and mTLS channel credentials.
 */
final class GrpcTransport implements TransportInterface
{
    public function __construct(private Config $config) {}

    public function availabilitySubmit(array $criteria): array
    {
        throw new TransportException('gRPC not wired yet. Generate stubs and implement.');
    }
    public function availabilityPoll(string $requestId, int $sinceSeq, int $waitMs): array
    {
        throw new TransportException('gRPC not wired yet. Generate stubs and implement.');
    }
    public function isLocationSupported(string $agreementRef, string $locode): bool
    {
        throw new TransportException('gRPC not wired yet. Generate stubs and implement.');
    }
    public function bookingCreate(array $payload, ?string $idempotencyKey = null): array
    {
        throw new TransportException('gRPC not wired yet. Generate stubs and implement.');
    }
    public function bookingModify(array $payload): array
    {
        throw new TransportException('gRPC not wired yet. Generate stubs and implement.');
    }
    public function bookingCancel(array $payload): array
    {
        throw new TransportException('gRPC not wired yet. Generate stubs and implement.');
    }
    public function bookingCheck(string $supplierBookingRef, string $agreementRef, ?string $sourceId = null): array
    {
        throw new TransportException('gRPC not wired yet. Generate stubs and implement.');
    }
}

File: sdks/java-agent/src/main/java/com/carhire/sdk/transport/GrpcTransport.java
package com.carhire.sdk.transport;

import com.carhire.sdk.Config;
import com.carhire.sdk.TransportException;

import java.util.Map;
import java.util.concurrent.CompletableFuture;

/**
 * gRPC transport – STUBS until proto files are generated and service clients are wired.
 * To generate stubs, run: mvn compile
 * Then implement methods by calling generated stubs with per-call deadlines and mTLS channel credentials.
 */
public class GrpcTransport implements TransportInterface {
    private final Config config;

    public GrpcTransport(Config config) {
        this.config = config;
    }

    @Override
    public CompletableFuture<Map<String, Object>> availabilitySubmit(Map<String, Object> criteria) {
        return CompletableFuture.failedFuture(
            new TransportException("gRPC not wired yet. Generate stubs and implement.")
        );
    }

    @Override
    public CompletableFuture<Map<String, Object>> availabilityPoll(String requestId, int sinceSeq, int waitMs) {
        return CompletableFuture.failedFuture(
            new TransportException("gRPC not wired yet. Generate stubs and implement.")
        );
    }

    @Override
    public CompletableFuture<Boolean> isLocationSupported(String agreementRef, String locode) {
        return CompletableFuture.failedFuture(
            new TransportException("gRPC not wired yet. Generate stubs and implement.")
        );
    }

    @Override
    public CompletableFuture<Map<String, Object>> bookingCreate(Map<String, Object> payload, String idempotencyKey) {
        return CompletableFuture.failedFuture(
            new TransportException("gRPC not wired yet. Generate stubs and implement.")
        );
    }

    @Override
    public CompletableFuture<Map<String, Object>> bookingModify(Map<String, Object> payload) {
        return CompletableFuture.failedFuture(
            new TransportException("gRPC not wired yet. Generate stubs and implement.")
        );
    }

    @Override
    public CompletableFuture<Map<String, Object>> bookingCancel(Map<String, Object> payload) {
        return CompletableFuture.failedFuture(
            new TransportException("gRPC not wired yet. Generate stubs and implement.")
        );
    }

    @Override
    public CompletableFuture<Map<String, Object>> bookingCheck(String supplierBookingRef, String agreementRef, String sourceId) {
        return CompletableFuture.failedFuture(
            new TransportException("gRPC not wired yet. Generate stubs and implement.")
        );
    }
}

